### DISCUSSION TITLE
Adding input_file_names to log_value / Adding input filename to Image Gen Info in Batch mode

### DISCUSSION THREAD

**Discussion by Chubbly** [Mikubill/sd-webui-controlnet#1809] [Ideas]
When running a batch of OpenPose images from a directory in txt2img, it can be difficult(/impossible) to recreate an image without knowing which pose file was used by ControlNet to generate the image.

Would it be possible to add the filename to PNG info when ControlNet is run in batch mode? I have looked through the code myself to see if I can create a PR but tracing how each image is used when in batch mode is difficult. The issue appears to be that the image gen info is created for the ControlNet unit and not for each image. But then the input file name is not in scope to append to extra_generation_params when the ControlNet is processing a single image. Am I reading this right or is there actually an easy way for this to be done?

**Comment by z4m1** [comment]
I had the same request and found this post that is similar, so I am posting it here.


Problem: Difficult to reproduce patterns created by CN

At present, the only way to reproduce a pattern is to remember the input image in one's head or to write it down manually.
It is not necessary for the input to be automatically entered by send t2i or other means, but at any rate, without the saved information, it is not possible to even reproduce it.

Request: Add an optional setting to record in PNG metadata or output information externally.
